src/main/webapp/mapScript.js

```diff
@@ -5,39 +5,88 @@ goog.require('templates.infoWindow');
 // Array that will hold the groups and there markers in a 2D Array
 // The array will allow the markers to be easilly all disabled as 
 // Groups were unchecked or checked 
-let groupArray = []; 
+let groupMarkers = new Map(); 
+
+// Default map placement centered at Mountain View, CA
+const DEFAULT_LAT = 37.3868;
+const DEFAULT_LNG = -122.085;
+const DEFAULT_ZOOM = 10;
+ 
+let map; 
 
 /**
  * Initialize the map to show Mountain View and events in that area
  */
 function initMap() {
-  
-    // Coordinates for the default map center of Mountain View, CA
-    const mapViewDefault = {lat: 37.3868, lng: -122.085}; 
+    const mapViewDefault = {lat: DEFAULT_LAT, lng: DEFAULT_LNG}; 
 
-    const map = new google.maps.Map(goog.dom.getElement('map'), {
-    // Zoom set to 8 as default until the radius is set up with events
-        zoom: 8,
+    map = new google.maps.Map(goog.dom.getElement('map'), {
+        // Zoom level will change automatically with user's search
+        zoom: DEFAULT_ZOOM,
         center: mapViewDefault,
     });
 
     const geocoder = new google.maps.Geocoder();
-    // TODO: repace the hardcoded event ID with 
-    getMarkerInfo(123, geocoder, map);
+    // TODO: repace the hardcoded event ID with actually value
+    getMarkerInfo('123', geocoder);
+
+
+    autoCompleteAndZoom(); 
+
     document.getElementById('submit').addEventListener('click', () => {
-        geocodeAddress(geocoder, map);
+        geocodeAddress(geocoder);
     });
+
 }
 
+/**
+ * Adds the auto Complete for the address search bar 
+ * Allows user to hit enter instead of button
+ * Automatically zooms according to the location inputed
+ */
+function autoCompleteAndZoom() {
+    const address = document.getElementById('address');
+    const searchBox = new google.maps.places.SearchBox(address);
+
+    map.addListener("bounds_changed", function() {
+        searchBox.setBounds(map.getBounds());
+    });
+
+    searchBox.addListener("places_changed", function() {
+        const places = searchBox.getPlaces();
+
+        // Checks if there are not autocomplete options
+        if (places.length == 0) {
```

Comment:
I'm not as familiar with js; are you able to do `places.isEmpty()` here?

Reply:
No, it doesn't work :(